mappymatch/constructs/trace.py

```diff
@@ -46,6 +46,17 @@ def __add__(self, other: Trace) -> Trace:
     def __len__(self):
         return len(self._frame)
 
+    def __str__(self):
+        output_lines = [
+            "Mappymatch Trace object",
+            f'coords_list: {self.coords_list if hasattr(self, "coords_list") else None}',
+            f'frame: {self._frame if hasattr(self, "_frame") else None}',
```

Comment:
It seems like checking if the trace has the `_frame` attribute is overkill as we don't ever expect a Trace without that attribute. Can you envision a scenario where we would actually not have the `_frame` attribute?

Reply:
Under normal use as the code is written today, no. However, objects in Python are mutable enough that it's difficult to ensure that a given object will have a given attribute. To use two examples that are extremely poor programming practice:

```
In [1]: from mappymatch.constructs.trace import Trace

In [2]: trace = Trace('foo')

In [3]: delattr(trace, '_frame')

In [4]: print(trace)
Mappymatch Trace object
coords_list: None
frame: None
```

Or: 

```
In [1]: from mappymatch.constructs.trace import Trace

In [2]: class InheritTrace(Trace):
   ...:     def __init__(self):
   ...:         pass
   ...: 

In [3]: trace = InheritTrace()

In [4]: print(trace)
Mappymatch Trace object
coords_list: None
frame: None
```

However, if not having a `_frame` is such an exceptional event that you WANT the entire program to raise an exception and notify a human, that is actually a good argument for not checking it.

Reply:
Changed in https://github.com/NREL/mappymatch/pull/135/commits/5895949f6527f1714db8e37efc1627beb680db03

Reply:
Yes, good point, as python is interpreted that case you mention would result in the trace not having the `_frame` attribute. I do think it makes sense to blow up if that attribute doesn't exist as it would be an exception (since you can't use a trace without a _frame attribute)